src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsReader.java

```diff
@@ -176,12 +189,28 @@ public void search(String field, byte[] target, KnnCollector knnCollector, Bits
      */
     @Override
     public void close() throws IOException {
+        // Clean up allocated vector indices resources from cache.
+        final NativeMemoryCacheManager nativeMemoryCacheManager = NativeMemoryCacheManager.getInstance();
+        // TMP
+        System.out.println(" &&&&&&&&&&&& [KDY] 990KV Cache manager BEFORE - " + nativeMemoryCacheManager.kdyPrintMap());
+        // TMP
+        indexToVectorFileName.values().forEach(nativeMemoryCacheManager::invalidate);
+
+        // Close a reader.
         IOUtils.close(flatVectorsReader);
+
+        // Clean up quantized state cache.
         if (quantizationStateCacheKeyPerField != null) {
+            final QuantizationStateCacheManager quantizationStateCacheManager = QuantizationStateCacheManager.getInstance();
             for (String cacheKey : quantizationStateCacheKeyPerField.values()) {
-                QuantizationStateCacheManager.getInstance().evict(cacheKey);
+                quantizationStateCacheManager.evict(cacheKey);
             }
         }
+
+        // TMP
+        System.out.println(" &&&&&&&&&&&& [KDY] 990KV::close() - " + indexToVectorFileName);
```

Comment:
This too!
After presenting the idea of removing FileWatcher, this logging lines will be deleted.